net/src/p2p_network.rs

```diff
@@ -49,6 +50,9 @@ impl P2pNetwork {
         };
         // Create NetConnectionThread with appropriate worker factory
         let connection = NetConnectionThread::new(handler, worker_factory, None)?;
+        if let P2pBackendKind::IPC = p2p_config.backend_kind {
+            sleep(Duration::from_millis(1000));
+        }
```

Comment:
I believe it would be nice to add a comment stating that this is a temporary hack until network init is properly handled.

Reply:
+1